validator/client/validator.go

```diff
@@ -1139,7 +1140,40 @@ func (v *validator) StartEventStream(ctx context.Context, topics []string, event
 	v.validatorClient.StartEventStream(ctx, topics, eventsChannel)
 }
 
-func (v *validator) ProcessEvent(event *eventClient.Event) {
+func (v *validator) checkDependentRoots(ctx context.Context, head *structs.HeadEvent, slot primitives.Slot) error {
+	if head == nil {
+		return errors.New("received empty head event")
+	}
+	prevDepedentRoot, err := bytesutil.DecodeHexWithLength(head.PreviousDutyDependentRoot, fieldparams.RootLength)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode previous duty dependent root")
+	}
+	currEpochStart, err := slots.EpochStart(slots.ToEpoch(slot))
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(prevDepedentRoot, v.duties.PrevDependentRoot) {
+		if err := v.UpdateDuties(ctx, currEpochStart); err != nil {
```

Comment:
do these update duties need any kind of deadline?

Reply:
our other ones have a slot deadline I guess

Reply:
good point, would you rather I create a new context with a deadline here?

Reply:
Maybe it makes sense to put the slot deadline in `run`, similar to `case slot := <-v.NextSlot()` condition

```
		case e := <-eventsChan:
			v.ProcessEvent(ctx, e)
```